addons/utm/models/utm_campaign.py

```diff
@@ -9,6 +9,7 @@ class UtmCampaign(models.Model):
     _description = 'UTM Campaign'
     _rec_name = 'title'
 
+    active = fields.Boolean('Active', default=True)
```

Comment:
When assigning UTM values from a link, there is a method that will automatically match the campaign name with the URL parameters.

So if you have:

www.odoo.com/shop?utm_campaign=christmas

It will automatically match a campaign named "christmas".

Should we maybe add a context "test_active=False" there then? To avoid having the campaign automatically re-created?
Should ideally be tested to make sure.

Reply:
I re-tested to make sure and indeed my explanation was a little bit incomplete :smile: 

To create a campaign you have to:
- Navigate to `http://127.0.0.1:8069/shop?utm_campaign=bloup` (localhost is not happy with that, something about setting a domain to cookie, did not look into it too much)
- Purchase a table from the shop (you don't have to pay for it but I think you have to proceed to the checkout and fill in information)
- You should then see a new campaign "bloup" pop up in the list

![image](https://user-images.githubusercontent.com/41282536/232006809-21b6522b-708e-4c56-83c4-df762b876873.png)

More details:

The "default_get" where we extract cookies and call the "_find_or_create_record" method for utm records is inside the utm.mixin, so you need to create an object that extends that mixin to trigger that code (a sale.order in this case).

Hope that clarifies it :slightly_smiling_face: 

And yes I think your code is correct, but ideally it should be tested (there are probably already tests about creating utm records automatically, could be an additional one there).